transmission/transmission_test.go

```diff
@@ -924,18 +924,25 @@ func TestHoneycombTransmissionFlush(t *testing.T) {
 			if atomic.CompareAndSwapInt32(&batchCount, 0, 1) {
 				return b
 			}
-			return &fakeBatch{}
 			// We want to be sure that the data we enqueue is flushed in the batch we
-			// expect. By not having s send channel on subsequent batches, we'll
+			// expect. By having a closed send channel on subsequent batches, we'll
 			// panic if we flush to other batches.
+			sendChan2 := make(chan []interface{})
+			close(sendChan2)
+			return &fakeBatch{send: sendChan2}
 		}
 
 		if err := w.Start(); err != nil {
 			t.Error("unable to start", err)
 		}
 		defer w.Stop()
 
+		// This call wants the Add to run first, and then make sure that the Flush doesn't lose the data.
+		// But on a fast machine, the Add might actually be interrupted by the Flush, and then correctly add the
+		// data to the *second* queue -- which is correct, but not what the test is looking for. So we'll
+		// give the Add a few msec to run first.
 		w.Add(ev)
+		time.Sleep(50 * time.Millisecond)
```

Comment:
I think we can use metrics to wait for the `Add` deterministically (wait for `messages_queued` to be the right number), but it's probably not worth the hassle of setting up metrics, eh?

Reply:
I actually tried that, but it failed because the metrics are updated *before* the item is added to the queue. So you can still see deadlocks sometimes. This way isn't pretty but I haven't had it fail yet.

Reply:
Rubbin' some sleep on it.

![tenor-243394000](https://user-images.githubusercontent.com/517302/148988254-5d5e8f4d-9d57-4260-b7ea-2dcd0f71c814.gif)